crates/rattler_repodata_gateway/src/fetch/mod.rs

```diff
@@ -440,11 +468,18 @@ pub async fn fetch_repo_data(
         cache_headers.add_to_request(&mut headers)
     }
     // Send the request and wait for a reply
-    let response = request_builder
-        .headers(headers)
-        .send()
-        .await?
-        .error_for_status()?;
+    let result = request_builder.headers(headers).send().await;
+    let response = match result {
```

Comment:
No need to introduce another variable.

```suggestion
    let response = match request_builder.headers(headers).send().await {
```

Reply:
Changed